sample-app/src/components/FilterSearch.tsx

```diff
@@ -0,0 +1,90 @@
+import { FilterSearchResponse } from "@yext/answers-headless";
+import { SearchParameterField } from "@yext/answers-headless";
+import { useRef, useState } from "react";
+import { useAnswersActions } from '@yext/answers-headless-react';
+import InputDropdown from "./InputDropdown";
+import renderWithHighlighting from "./utils/renderWithHighlighting";
+import { Option } from "./Dropdown";
+
+const SCREENREADER_INSTRUCTIONS = 'When autocomplete results are available, use up and down arrows to review and enter to select.'
+
+export interface FilterSearchProps {
+  title: string,
+  sectioned: boolean,
+  searchFields: SearchParameterField[],
+  screenReaderInstructionsId: string
+}
+
+export default function FilterSearch (props: FilterSearchProps): JSX.Element {
+  const { title, sectioned, searchFields, screenReaderInstructionsId } = props;
+  const answersActions = useAnswersActions();
+  const [input, setInput] = useState('');
+  const [results, updateResults] = useState<FilterSearchResponse|undefined>();
+  const requestId = useRef(0);
+  const responseId = useRef(0);
+
+  async function executeFilterSearch (inputValue: string) {
+    const currentId = ++requestId.current;
+    const results = await answersActions.executeFilterSearch(inputValue, sectioned, searchFields);
+    if (currentId >= responseId.current) {
+      responseId.current++;
+      updateResults(results);
+    }
+  }
+
+  let options: {results: Option[], label?: string}[] = [];
+  if (results) {
+    results.sections.forEach(section => {
+      let results: Option[] = [];
+      section.results.forEach(result => {
+        results.push({
+          value: result.value,
+          render: () => renderWithHighlighting(result)
+        });
+      });
+      options.push({
+        results: results,
+        label: section.label
+      });
+    });
+  }
+
+  return (
+    <div className='FilterSearch'>
+      <h1>{title}</h1>
+      <InputDropdown
+        inputValue={input}
+        placeholder='this is filter search...'
+        screenReaderInstructions={SCREENREADER_INSTRUCTIONS}
+        screenReaderInstructionsId={screenReaderInstructionsId}
+        options={options}
+        optionIdPrefix='Autocomplete__option'
+        onlySubmitOnOption={true}
+        onSubmit={(optionValue, sectionIndex, optionIndex) => {
+          if (sectionIndex !== undefined && optionIndex !== undefined && results) {
+            const option = results.sections[sectionIndex].results[optionIndex];
+            if (option.filter) {
+              answersActions.setFilterOption({ ...option.filter, selected: true });
+              answersActions.executeVerticalQuery();
+            }
+          }
+        }}
+        updateInputValue={newInput => {
+          setInput(newInput);
+        }}
+        updateDropdown={(input) => {
+          executeFilterSearch(input);
+        }}
+        cssClasses={{
+          optionContainer: 'Autocomplete',
```

Comment:
Would it be possible to make all of the "Autocomplete" prefixed css classes the default values? Or maybe we don't even need props for them, at least not yet?.

Reply:
I was going off of the css classes in the SearchBar component, but I can remove the props from both components, or change them to be optional with the `Autocomplete` prefixed classes as the default in the Dropdown component. It would just depend on whether we want to require or even support custom css class names for that

Reply:
ah yeah I'm kind of scope creeping here, we can keep the classes for now if you like (also Connor is doing some css stuff right now so maybe that's for the best)

Reply:
ok, I think I'll keep it as is for now then